WordPress/gutenberg
# DataViews: Mark the new Templates pages as stable

Related #55083 

 - This PR stabilizes the new templates page based on data views as table and replaces entirely the old page. 
 - It keeps the "view" menu hidden behind the experimental flag because the "side by side" and "grid views" are not ready yet.
 - It would be good to replace the template parts page as well (which will allow us to remove a bunch of other components).

The idea here is to start getting some real usage feedback about the dataviews, to be completely ready when we reach the beta1 for the new WP release.

**Testing instructions**

 - Disabled the "data views" experiment.
 - Open the "browse all templates" page in the site editor.
 - It should show the new data views page without the "view" menu.
